AppServicesUsageSamples/apps/property-encryption/README.md

```diff
@@ -0,0 +1,83 @@
+# Property level encryption
+The goal of property-level encryption is to ensure that only the users themselves can access their data and prevent any unauthorized access by other parties, including MongoDB. 
+
+Realm does not have built-in property-level encryption support. However, with this sample we will demonstrate how to add this feature with the Realm Sync and Android Keystore APIs. Furthermore, this solution would enable users to access their data simultaneously from multiple devices. All of this is achieved by leveraging the powerful capabilities of MongoDB services.
+
+## Introduction
+Property-level encryption restricts access to sensitive data by applying a ciphering algorithm with a unique key that will encrypt the data on the device. The encrypted data will be synced to Atlas, thus only allowing people with the proper key to be able to read it again. This also includes MonogDB itself.
+
+It is crucial to ensure that only the rightful users have access to the keys.
+
+## Considerations
+The purpose of this demo is to show how can we leverage MongoDB and other tools to provide end-to-end property-level encryption to users with multi-device access.
+
+There are different algorithms available to achieve encryption, the ones selected in this sample are for demonstrative purposes. Any production development must select them based on their security requirements. 
+
+## Key handling
+Keys are stored in a key store on the device, a secure repository that safe-keeps the keys and guarantees that only the rightful users have access to them. It normally has some hardware-enhanced security features.
+
+For this demo, we will use the Android Keystore available from SDK version > 18. It offers hardware-enhanced security like biometric authentication and guarantees that the cryptographic material stored in it is not exportable.
+
+We will use the symmetric encryption algorithm AES, as it provides great performance when encrypting files and data.
+
+## Multi-device support
+As we mentioned, the local device keystore does not support exporting keys, this means that a user would not be able to access their data from a new device as the encryption keys would only be available from the original one. To overcome this we need a shared keystore that would allow accessing these keys from any device. 
+
+![alt text](diagram1.svg "Deployment")
+
+For this sample, we store the user keys in a password-protected Keystore in the user's `CustomData`. This allows you to control access to these (encrypted) keys based on the permission system in App Services.
```

Comment:
Should we reference the `initializeCustomData.js` and `updateKeyStore.js` functions here to give some direction of where this is handled?

Reply:
Great point!